crates/engine/tree/src/tree/payload_processor/multiproof.rs

```diff
@@ -684,8 +706,59 @@ pub(super) struct MultiProofTask {
     max_targets_for_chunking: usize,
 }
 
+/// Context for multiproof message batching loop.
+///
+/// Contains processing state that persists across loop iterations.
+struct MultiproofBatchCtx {
```

Comment:
I believe these new types are defined out of order and should be moved below the `impl MultiProofTask {` block